core/src/main/scala/org/apache/spark/status/protobuf/RDDOperationGraphWrapperSerializer.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.status.protobuf
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.rdd.DeterministicLevel
+import org.apache.spark.status.{RDDOperationClusterWrapper, RDDOperationGraphWrapper}
+import org.apache.spark.ui.scope.{RDDOperationEdge, RDDOperationNode}
+
+class RDDOperationGraphWrapperSerializer extends ProtobufSerDe {
+
+  override val supportClass: Class[_] = classOf[RDDOperationGraphWrapper]
+
+  override def serialize(input: Any): Array[Byte] = {
+    serialize(input.asInstanceOf[RDDOperationGraphWrapper])
+  }
+
+  private def serialize(input: RDDOperationGraphWrapper): Array[Byte] = {
+    val builder = StoreTypes.RDDOperationGraphWrapper.newBuilder()
+    builder.setStageId(input.stageId)
+    input.edges.foreach { e =>
+      builder.addEdges(serializeRDDOperationEdge(e))
+    }
+    input.outgoingEdges.foreach { e =>
+      builder.addOutgoingEdges(serializeRDDOperationEdge(e))
+    }
+    input.incomingEdges.foreach { e =>
+      builder.addIncomingEdges(serializeRDDOperationEdge(e))
+    }
+    builder.setRootCluster(serializeRDDOperationClusterWrapper(input.rootCluster))
+    builder.build().toByteArray
+  }
+
+  def deserialize(bytes: Array[Byte]): RDDOperationGraphWrapper = {
+    val wrapper = StoreTypes.RDDOperationGraphWrapper.parseFrom(bytes)
+    new RDDOperationGraphWrapper(
+      stageId = wrapper.getStageId,
+      edges = wrapper.getEdgesList.asScala.map(deserializeRDDOperationEdge).toSeq,
```

Comment:
I found that there are many redundant `toSeq` for Scala 2.12 in `status.protobuf` package,  this is for Scala 2.13 compatibility due to `Seq` represents `collection.Seq` in Scala 2.12 and `immutable.Seq` in Scala 2.13.

This conversion will not affect Scala 2.12, but will make the performance of Scala 2.13 worse than Scala 2.12. Since these are internal definitions of Spark, I suggest explicitly defining them as `scala.collection.Seq` to make no performance difference  between  Scala 2.12 and Scala 2.13. Do you think it's ok? @gengliangwang

Reply:
@LuciferYang Agree, and since these are private[spark], it shouldn't be an issue.

Reply:
> explicitly defining them as scala.collection.Seq to make no performance difference

@LuciferYang could you explain details?

Reply:
https://github.com/apache/spark/pull/39215/files is doing some refactor work, which does not block the current pr